### PR TITLE
WIP unibilium recipe

### PR DESCRIPTION
**Pull request by miqlas**
Work in Progress.
Now i get : `undefined reference to`__stack_chk_fail_local'`

Help required.

### CODE REVIEW

**Comment by miqlas** [comment]
Have anybody idea, what am i doing wrong? It compiles without a hitch outside of chroot...

**Comment by fbrosson** [comment]
Maybe the recipe is missing a dependency.
Does «**`haikuporter --nochroot unibilium`**» work ?
BTW, «**`--nochroot`**» has a one-letter equivalent, «**`-C`**», so you can use that instead of «**`--nochroot`**».

File: dev-libs/unibilium/unibilium-1.2.0.recipe [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtool$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	pkgconfig$secondaryArchSuffix
```
Comment (fbrosson) [inline dev-libs/unibilium/unibilium-1.2.0.recipe]: Should be:
```
	cmd:pkg_config$secondaryArchSuffix
```

**Comment by fbrosson** [comment]
Oops, «**`haikuporter --nochroot ...`**» is not working for me. Maybe I need to upgrade something on my system.
Anyway, here's how you might identify a missing dependency:
* Re-build unibilium without haikuporter (or skip this step if you still have your build directory around), since you said it compiles without a hitch when you build without haikuporter.
* Then use «**`objdump -x /path/to/libunibilium.so | grep NEEDED`**» to see which shared libraries it needs.
* Use «**`pkgman search lib:libNAME`**» for every **`libNAME.so.X`** that you get with the previous command.
* Add one «**`lib:libNAME`**» for each library that is being provided by a package.

BTW, I havent tried this myself, so I'm not sure it works. Good luck!

**Comment by miqlas** [comment]
fbrosson, the only objdump output is:
`NEEDED      libroot.so`

Now, what?

**Comment by fbrosson** [comment]
This means libunibilium.so does not depend on any lib provided by other recipes. So the runttime dependencies seem OK.
Maybe you could compare the output of the configure script in the chrooted environment versus the non-chrooted environment.

File: dev-libs/unibilium/unibilium-1.2.0.recipe [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	cmd:ld$secondaryArchSuffix
+	cmd:libtool$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+export LDFLAGS="-lnetwork -lbsd -lroot"
+export CFLAGS="-D_BSD_SOURCE"
```
Comment (waddlesplash) [inline dev-libs/unibilium/unibilium-1.2.0.recipe]: These should be in `BUILD`, no?

**Comment by waddlesplash** [comment]
@miqlas Travis build failed; please check.

**Comment by miqlas** [comment]
Strange, but it builds just fine outside of Haikuporter, but inside it cannot find -lubsan and -lasan
Grepped the whole tree, nothing.
Must be a bug in the matrix or something.

**Comment by miqlas** [comment]
Ok, it actually works now.

**Comment by miqlas** [comment]
Merge?

**Pull request merged**